lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1633,27 +1650,96 @@ LogicalResult PaddedSharedEncodingAttr::verify(
 
-  if (order.empty())
-    return emitError() << "order cannot be empty";
+  const auto &ll = linearComponent;
+  // The linear layout should mape from [offset, block] to [dim0..dimN). All
+  // bases should be 0 or power of twos and move in a single direction without
+  // broadcasting
+  for (auto outDim : ll.getInDimNames()) {
+    if (outDim.str() != "offset" && outDim.str() != "block")
+      return emitError() << "linearComponent must have `offset` and `block` as "
+                            "only input dims, found: "
+                         << outDim.str();
+  }
+  if (ll.getNumInDims() != 2)
+    return emitError() << "linearComponent must have `offset` and `block` as "
+                          "only input dims";
 
-  if (order.size() != ctaLayout.getRank())
-    return emitError() << "order size (" << order.size()
-                       << ") must match CTALayout rank (" << ctaLayout.getRank()
-                       << ")";
-  return verifyLayoutOrder(emitError, order);
+  // outDims are ['dim0', 'dim1', ...]
+  for (auto [i, dim] : llvm::enumerate(ll.getOutDimNames())) {
+    if (dim.str() != ("dim" + llvm::Twine(i)).str()) {
+      return emitError()
+             << "Expected output dimensions to be ['dim0', 'dim1', ...]. Got "
+             << dim << " at position " << i;
+    }
+  }
+
+  const auto &bases = ll.getBases();
+
+  auto nonZero = [](auto val) { return val != 0; };
+  for (const auto &dimBases : llvm::make_second_range(bases)) {
+    if (!llvm::all_of(dimBases, [&](const auto &basis) {
+          return llvm::count_if(basis, nonZero) <= 1;
+        })) {
+      return emitError()
+             << "Each offset basis must move in at most one dimension.";
+    }
+    if (!llvm::all_of(dimBases, [&](const auto &basis) {
+          return llvm::all_of(
+              basis, [](auto v) { return v == 0 || llvm::isPowerOf2_32(v); });
+        })) {
+      return emitError() << "Each offset basis must be 0 or a power of two.";
+    }
+    if (!llvm::all_of(dimBases, [&](const auto &basis) {
+          return llvm::count_if(basis, nonZero) != 0;
+        })) {
+      return emitError() << "Broadcasting offset basis are not supported.";
+    }
+  }
+
+  return success();
 }
 
 PaddedSharedEncodingAttr PaddedSharedEncodingAttr::get(
     MLIRContext *context, ArrayRef<std::pair<unsigned, unsigned>> intervalPads,
-    ArrayRef<unsigned> order, CTALayoutAttr ctaLayout) {
+    ArrayRef<unsigned> order, ArrayRef<int64_t> shape,
+    CTALayoutAttr ctaLayout) {
+  auto rank = shape.size();
+  auto outDimNames = standardOutDimNames(context, rank);
+  StringAttr kOffset = StringAttr::get(context, "offset");
+
+  // Create identity mapping based on shape and order
+  LinearLayout linearComponent;
+  for (auto dim : order) {
+    linearComponent *=
+        LinearLayout::identity1D(shape[dim], kOffset, outDimNames[dim]);
+  }
+  // Transpose to standard out dim order
+  linearComponent = linearComponent.transposeOuts(outDimNames);
```

Comment:
All this is implemented in `identityStandardND`. Also I believe the transposeOuts is already done in `combineCTACgaWithShape.